lib/errors.js

```diff
@@ -0,0 +1,33 @@
+'use babel';
+
+//
+// Utility functions for parsing errors
+//
+
+const level2severity = (level) => {
+  switch (level) {
+    case 'warning': return 'warning';
+    case 'error': return 'error';
+    case 'note': return 'info';
+    case 'help': return 'info';
+    default: return 'error';
+  }
+};
+
+const level2type = (level) => {
+  return level.charAt(0).toUpperCase() + level.slice(1);
+};
+
+// Set location for special cases when the compiler doesn't provide it
+function preprocessMessage(msg, buildWorkDir) {
+  if (msg.file) {
+    return true;
+  }
+  if (msg.message === 'main function not found') {
+    msg.file = buildWorkDir + '/src/main.rs';	// TODO: When running an example, set the path to the example file.
```

Comment:
The thing is that the path can be customised, with the following `Cargo.toml` it is src/foo.rs and not src/main.rs.

`[[bin]]
name = "foo"
path = "src/foo.rs"`

You can completely overwrite this behaviour and even define multiple binaries. In these edge cases the guess will be wrong and point to a nonexistent or not the right file. Pointing to the wrong binary might be especially confusing if that one has a main function...

The question is, is it worth to have better diagnostics in 97℅ of the cases if that means getting 3℅ completely wrong? These numbers are just my personal estimate and may be far off.